### PR TITLE
[MRG][BRANCH-0.4.x] Regression on pickling classes from the __main__ module 

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR backports https://github.com/cloudpipe/cloudpickle/pull/132 to branch-0.4.x and fixes https://github.com/cloudpipe/cloudpickle/issues/131.

### CODE REVIEW

File: tests/cloudpickle_test.py [hunk 1]
```diff
@@ -736,6 +737,62 @@ def test_builtin_type__new__(self):
...
+
+        cloned = subprocess_pickle_echo(f2)
+        assert cloned(2) == f2(2)
+
+        cloned = subprocess_pickle_echo(f3)
+        assert cloned() == f3()
+
+        cloned = subprocess_pickle_echo(f4)
+        assert cloned(2) == f4(2)
+        """
```
Comment (HyukjinKwon) [inline tests/cloudpickle_test.py]: format and protocol was removed out here comparing to the original.

**Comment by HyukjinKwon** [comment]
Seems the test failure is by https://github.com/cloudpipe/cloudpickle/pull/151.

File: tests/cloudpickle_test.py (hunk 1 above)
Comment (HyukjinKwon) [inline tests/cloudpickle_test.py]: I removed protocol comparing to the original

File: tests/testutils.py [hunk 2]
```diff
@@ -28,9 +29,13 @@ def subprocess_pickle_echo(input_data):
 
     """
     pickled_input_data = dumps(input_data)
```
Comment (HyukjinKwon) [inline tests/testutils.py]: ditto

**Comment by rgbkrk** [comment]
Since I merged the other, I think you can rebase this branch now.

**Pull request merged**